crates/vm/src/builtins/bool.rs

```diff
@@ -78,18 +79,46 @@ impl PyObjectRef {
 }
 
 #[pyclass(name = "bool", module = false, base = PyInt)]
-pub struct PyBool;
+#[repr(transparent)]
+pub struct PyBool(pub PyInt);
 
 impl PyPayload for PyBool {
     #[inline]
     fn class(ctx: &Context) -> &'static Py<PyType> {
         ctx.types.bool_type
     }
+
+    /// PyBool reuses PyInt's TypeId
+    #[inline]
+    fn payload_type_id() -> std::any::TypeId {
+        std::any::TypeId::of::<PyInt>()
+    }
```

Comment:
This is okay here, because PyInt and PyBool shares same payload. The only difference is type.

But implementing this makes `PyPayload::downcastable` to be always `true` for any kind of `PyInt`. It doesn't make failing tests on this patch because we don't use that pattern. But it obviously will be a sort of trap for future developing.

Another question is that `PyPayload::downcastable` only takes &PyObject currently. We can't compare it to actual type... because we don't have vm.ctx.